### DISCUSSION TITLE
Play audio immediately after connecting

### DISCUSSION THREAD

**Discussion by jokubasver** [pschatzmann/ESP32-A2DP#145] [Ideas]
I think a useful feature to have is to autoplay music after connecting to a phone. This is handy if you are using A2DP as a bluetooth aux device in a car - phone connects and immediately starts playing music, or if you walk away from the car and later come back - the audio resumes from where it was left off. Here's an example of how I have implemented this in a sketch: https://github.com/jokubasver/saab-93NG-bluetooth-aux/commit/dd12b775c7b2f3dd0baf54b07b4c4c4792770c4f

Maybe such a feature could be implemented in the library.

**Comment by pschatzmann** [comment]
Thanks for your feed-back, but I don't think there is any need to change any code and it should be pretty easy to implement this in a sketch. Just use the connection state callback method (set_on_connection_state_changed) and call the play method when you get a connected notification.

Or am I missing anything ?

**Reply by jokubasver** [reply]
Yes, that works too. I was just thinking that this could be implemented into, let's say `set_auto_reconnect` as a parameter, to make it simpler and quicker.
